osu.Game.Rulesets.Osu/Objects/Drawables/DrawableRepeatPoint.cs

```diff
@@ -71,6 +77,14 @@ protected override void UpdateCurrentState(ArmedState state)
             }
         }
 
-        public void UpdateSnakingPosition(Vector2 start, Vector2 end) => Position = repeatPoint.RepeatIndex % 2 == 0 ? end : start;
+        public void UpdateSnakingPosition(Vector2 start, Vector2 end)
+        {
+            Position = isRepeatAtEnd ? end : start;
+            var curve = drawableSlider.Body.CurrentCurve;
+            if (curve.Count < 3 || curve.All(p => p == Position))
+                return;
+            var referencePoint = curve[isRepeatAtEnd ? curve.IndexOf(Position, curve.Count - 2) - 1 : curve[0] == curve[1] ? 2 : 1];

```

Comment:
Can we split this out into two loops? Something like:

```
Vector2 referencePoint;
if (isRepeatAtEnd)
    referenceIndex = curve.Count - 1;
    while (curve[referenceIndex] == Position)
        referenceIndex--;
else
    referenceIndex = 0;
    while (curve[referenceIndex] == Position)
        referenceIndex++;
```

With appropriate comments describing that this is finding the next point that is not equal to Position? Had to mull over this for quite a while to figure out what it did, and what these seemingly random offsets did.

This will also fix the case where there are more than 2 points with the same value in the false case.